src/client/network/nusight_network.ts

```diff
@@ -33,16 +87,16 @@ export class NUsightNetwork {
   public onNUClearMessage<T>(messageType: MessageType<T>, cb: MessageCallback<T>) {
     const messageTypeName = this.messageTypePath.getPath(messageType)
     return this.nuclearnetClient.on(`NUsight<${messageTypeName}>`, (packet: NUClearNetPacket) => {
-      // Remove NUsight header for decoding by protobufjs
-      const buffer = new Uint8Array(packet.payload).slice(HEADER_SIZE)
-      const message = messageType.decode(buffer)
```

Comment:
I have no idea how, but you'll need to pass `messageType.decode` to the worker somehow, because the decoding process is different per message... 😞